tilings/tiling.py

```diff
@@ -1838,6 +1838,15 @@ def __ne__(self, other: object) -> bool:
             or self.assumptions != other.assumptions
         )
 
+    def __contains__(self, gp: GriddedPerm) -> bool:
```

Comment:
This would be better more like:
gp.avoids(*obs) and all(gp.contains(*req) for req in reqs)

Reply:
Absolutely, this is much cleaner.